### ISSUE TITLE
NVDA freeze when copying large amount of text

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#11843]
### Steps to reproduce:

* go to Wikipedia's [World War I](https://en.wikipedia.org/wiki/World_War_I) page
* press ctrl+A to select all
* press ctrl+C to copy
* Alt+Tab to another application

### Actual behavior:

NVDA freezes during a lot of time (maybe 30 seconds)

### Expected behavior:
Text should be copied without NVDA freezing.

### Note
Issue most likely due to the merge of PR #9843.
A solution could be to report the copied text only if its length is less than a specific value, and to report the number of copied characters else. This is the strategy used by the selection reporting.

#### NVDA installed/portable/running from source:
Temporary copy (installer) and portable version.
#### NVDA version:
Today's alpha:
Version : alpha-21371,376edfcf

#### Windows version:
Windows 10 1809 (64-bit) build 17763.1518
#### Name and version of other software in use when reproducing the issue:
Chrome - Version 85.0.4183.121 (Build officiel) (64 bits)

#### Does the issue still occur after restarting your computer?
Not tested.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA 2020.3 is OK.

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by CyrilleB79** [comment]
Cc @JulienCochuyt

**Comment by JulienCochuyt** [comment]
@CyrilleB79 
Sorry for missing this issue while testing.
My primary speech synth is eSpeak, which does not exhibit the issue. I've thought Speech Refactor was the cause for the success.
Now testing with OneCore proves me wrong.
You're proposed solution of aligning to the behavior of selection announcement / clipboard announcement sound reasonable to me.
I'll try to quickly work it out this afternoon.

**Comment by JulienCochuyt** [comment]
`speech._getSelectionMessageSpeech` limits to 512 characters and simply reports "%d characters"
`globalCommands.script_reportClipboardText` limits to 1024 characters and reports "The clipboard contains a large portion of text. It is %s characters long"

I'd rather go the middle way: Limit to 1024 and simply report "%d characters copied to clipboard".
An alternative approach could be to also report portions of the beginning and end of the lengthy text, ie. "%d characters copied to clipboard, starting with {snippet}, ending with {snippet}".
What do you think?

**Comment by CyrilleB79** [comment]
> `speech._getSelectionMessageSpeech` limits to 512 characters and simply reports "%d characters"
> `globalCommands.script_reportClipboardText` limits to 1024 characters and reports "The clipboard contains a large portion of text. It is %s characters long"
> 
> I'd rather go the middle way: Limit to 1024 and simply report "%d characters copied to clipboard".

I agree with this solution.

> An alternative approach could be to also report portions of the beginning and end of the lengthy text, ie. "%d characters copied to clipboard, starting with {snippet}, ending with {snippet}".
> What do you think?

For now, I would suggest to limit to your first proposal to be more or less aligned with `speech._getSelectionMessageSpeech` and `globalCommands.script_reportClipboard` in order to fix the initial PR #9843.

Announcing the beginning and the end of the copied text could be done in a second time. I think there is a discussion about this regarding selection. If I find the issue again, I will add the reference here.
